lib/java/src/org/apache/thrift/async/TAsyncMethodCall.java

```diff
@@ -225,8 +231,13 @@ private void cleanUpAndFireCallback(SelectionKey key) {
     key.interestOps(0);
     // this ensures that the TAsyncMethod instance doesn't hang around
     key.attach(null);
```

Comment:
Adding cancel won't (doesn't) work.  The key is associated with the single underlying channel (wrapped in a `TNonblockingTransport`) from the `TAsyncClient` which is intended for use with multiple concurrent calls afaict.  The `org.apache.thrift.async.TestTAsyncClientManager#testParallelCalls` test uses the client in this way - 1 client, multiple concurrent calls - and demonstrates that cancelling the key cancels the underlying channel and, for example, you get 1 successful call and 4999 failed calls in the 5000 call test case as a result.